application/models/Survey.php

```diff
@@ -835,6 +835,33 @@ public function getSurveyTemplateConfiguration()
         return TemplateConfiguration::getInstance(null, null, $this->sid);
     }
 
+    /**
+     * Returns the name of the template to be used for the survey.
+     * It resolves inheritance from group and from default settings.
+     * 
+     * @return string
+     * 
+     * @todo:  Cache this on a private attribute?
+     */
+    public function getTemplateEffectiveName()
+    {
+        // Fetch template name from model
+        // This was already filtered on afterFind, so if the one at load time is not valid, will be replaced by default one
+        // If it is "inherit", means it will inherit from group, so we will replace it.
+        $sTemplateName = $this->template;
+
+        // if it is "inherit", get template name form group
+        if ($sTemplateName == 'inherit') {            
+            if (!empty($this->oOptions->template)) {
+                $sTemplateName = $this->oOptions->template;
+            } else {
+                throw new CException("Unable to get a template name from group for survey {$this->sid}");
```

Comment:
I like the new function :+1: .

But here : i don't know how it work for parent : parent can not be inherit too ? Then get it for "global" ?

Reply:
I actually tooke the code from digging into your example.
The code is from here: https://github.com/gabrieljenik/LimeSurvey/blob/65061857c0ae8514300746b481285e0004a7d576/application/models/TemplateConfiguration.php#L259

So, I think survey group inheritance is there. 
Now that you mention it, not sure how it would work with multiple survey groups inherted.
Is that possible? Does the TemplateConfiguration code supports it?

@Shnoulle What I don't like that much your intial approach (sorry :) ) is that there is a lot of processing (template instancing) just for getting the template name.

Reply:
Thanks : never look at getInstanceFromSurveyId :)